command/ca/renew.go

```diff
@@ -0,0 +1,458 @@
+	"github.com/smallstep/cli/ui"
+	"github.com/smallstep/cli/utils"
+	"github.com/urfave/cli"
+)
+
+func renewCertificateCommand() cli.Command {
+	return cli.Command{
+		Name:   "renew",
+		Action: command.ActionFunc(renewCertificateAction),
+		Usage:  "renew a valid certificate",
+		UsageText: `**step ca renew** <crt-file> <key-file>
+		[**--ca-url**=<uri>] [**--root**=<file>]
+		[**--out**=<file>] [**--expires-in**=<duration>] [**--force**]`,
+		Description: `
+**step ca renew** command renews the given certificate (with a request to the
+certificate authority) and writes the new certificate to disk - either overwriting
+<crt-file> or using a new file when the **--out**=<file> flag is used.
+
+With the **--daemon** flag the command will periodically update the given
+certificate. By default, it will renew the certificate before 2/3 of the validity
+period of the certificate has elapsed. A random jitter is used to avoid multiple
+instances running at the same time. The amount of time between renewal and
+certificate expiration can be configured using the **--expires-in** flag, or a
+fixed period can be set with the **--renew-period** flag.
+
+The **--daemon** flag can be combined with **--pid**, **--signal**, or **--exec**
+to provide certificate reloads on your services.
+
+## POSITIONAL ARGUMENTS
+
+<crt-file>
+:  The certificate in PEM format that we want to renew.
+
+<key-file>
+:  They key file of the certificate.
+
+## EXAMPLES
+
+Renew a certificate with the configured CA:
+'''
+$ step ca renew internal.crt internal.key
+Would you like to overwrite internal.crt [Y/n]: y
+'''
+
+Renew a certificate without overwriting the previous certificate:
+'''
+$ step ca renew --out renewed.crt internal.crt internal.key
+'''
+
+Renew a certificate forcing the overwrite of the previous certificate:
+'''
+$ step ca renew --force internal.crt internal.key
+'''
+
+Renew a certificate providing the <--ca-url> and <--root> flags:
+'''
+$ step ca renew --ca-url https://ca.smallstep.com:9000 \
+  --root /path/to/root_ca.crt internal.crt internal.key
+Would you like to overwrite internal.crt [Y/n]: y
+'''
+
+Renew skipped because it was too early:
+'''
+$ step ca renew --expires-in 8h internal.crt internal.key
+certificate not renewed: expires in 10h52m5s
+'''
+
+Renew the certificate before 2/3 of the validity has passed:
+'''
+$ step ca renew --daemon internal.crt internal.key
+'''
+
+Renew the certificate before 8 hours and 30m of the expiration time:
+'''
+$ step ca renew --daemon --expires-in 8h30m internal.crt internal.key
+'''
+
+Renew the certificate every 16h:
+'''
+$ step ca renew --daemon --renew-period 16h internal.crt internal.key
+'''
+
+Renew the certificate and reload nginx:
+'''
+$ step ca renew --daemon --exec "nginx -s reload" internal.crt internal.key 
+'''
+
+Renew the certificate and convert it to DER:
+'''
+$ step ca renew --daemon --renew-period 16h \
+  --exec "step certificate format --force --out internal.der internal.crt" \
```

Comment:
Why use --force if --out is different from the original file?

Reply:
because this is going to run periodically multiple times